### PR TITLE
Define `real` and `imag` for StaticInteger

### PR DESCRIPTION
**Pull request by chriselrod**
https://github.com/JuliaLinearAlgebra/Octavian.jl/pull/147

### CODE REVIEW

File: src/Static.jl [hunk 1]
```diff
@@ -430,6 +430,9 @@ Base.any(::Tuple{Vararg{True}}) = true
 Base.any(::Tuple{Vararg{Union{True, False}}}) = true
 Base.any(::Tuple{Vararg{False}}) = false
 
+Base.real(@nospecialize(x::StaticInteger)) = x
```
Comment (Tokazama) [inline src/Static.jl:433]:
```suggestion
Base.real(@nospecialize(x::StaticNumber)) = x
Base.real(@nospecialize(T::Type{<:StaticNumber})) = eltype(T)
```

This should avoid issues with type promotion.

File: src/Static.jl [hunk 2]
```diff
@@ -430,6 +430,9 @@ Base.any(::Tuple{Vararg{True}}) = true
...
+Base.imag(@nospecialize(_::StaticInteger)) = Zero()
```
Comment (Tokazama) [inline src/Static.jl:434]:
```suggestion
Base.imag(@nospecialize(x::StaticNumber)) = zero(x)
```
This allows it to work for all static numbers.

**Comment by chriselrod** [comment]
I formatted it locally with JuliaFormatter v1.0.5, but it fails the format test on CI using the same version of the formatter.

**Review by Tokazama — APPROVED** [review]

**Comment by Tokazama** [comment]
> I formatted it locally with JuliaFormatter v1.0.5, but it fails the format test on CI using the same version of the formatter.

Are we supposed to do this using specific SciML settings? I have to admit I haven't been keeping up with dev docs lately.

**Comment by chriselrod** [comment]
> > I formatted it locally with JuliaFormatter v1.0.5, but it fails the format test on CI using the same version of the formatter.
> 
> Are we supposed to do this using specific SciML settings? I have to admit I haven't been keeping up with dev docs lately.

That should happen automatically
https://github.com/SciML/Static.jl/blob/master/.JuliaFormatter.toml
Formatting passes now. I don't know what happened.
Merging master and reformatting fixed it.

**Pull request merged**